### ISSUE TITLE
Accessing the timestamp on the LogStash::Event instance create unnecessary new object

### ISSUE THREAD

**Issue by ph** [elastic/logstash#7318]
While debugging https://github.com/logstash-plugins/logstash-codec-rubydebug/pull/14, I have found that we create a new instance everytime that we call `Event#get("@timestamp")`, Since we use that instance in multiple places we should cache that value.

```
2.3.0 :002 > e = LogStash::Event.new({ "message" => "hello"})
 => #<LogStash::Event:0x3422b621>
2.3.0 :003 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x6e9514b4>
2.3.0 :004 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x1ff75c41>
2.3.0 :005 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x549a4c80>
2.3.0 :006 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x33def139>
2.3.0 :007 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x166eb657>
2.3.0 :008 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x10961656>
2.3.0 :009 > e.get("@timestamp")
 => #<LogStash::Timestamp:0x26d46bd3>
```

**Comment by original-brownbear** [comment]
Just linking this one:

But ... this would obviously go away automatically if we drop the use of `BiValue` altogether as suggested here: https://github.com/elastic/logstash/issues/7712

**Comment by original-brownbear** [comment]
fixed in #8334
